hs-ucan/library/Crypto/Key/Asymmetric/Public/Types.hs

```diff
@@ -79,3 +94,19 @@ instance ToSchema Public where
       & description ?~ "A public key"
       & NamedSchema (Just "PublicKey")
       & pure
+
+
+
+-- CONSTRUCTORS
+
+
+fromKeyType :: Text -> Text -> Either Text Public
+fromKeyType "Ed25519"   = fmap Ed25519PublicKey . parseUrlPiece
+fromKeyType "RSA"       = fmap RSAPublicKey . parseUrlPiece
+fromKeyType "Secp256k1" = fmap Secp256k1PublicKey . parseUrlPiece
+fromKeyType _           = \_ -> Left "Invalid Public.Key type"
```

Comment:
I tried to do this dynamically with the `Data.Data` module, but turned into a bit of hassle, so I kept it simple.